File: app/src/main/java/eu/indiewalkabout/mathbrainer/statistics/Results.java
package eu.indiewalkabout.mathbrainer.statistics;

import eu.indiewalkabout.mathbrainer.util.AppExecutors;
import eu.indiewalkabout.mathbrainer.util.SingletonProvider;

public class Results {

    /**
     * ---------------------------------------------------------------------------------------------
     * Init game results in db
     * ---------------------------------------------------------------------------------------------
     */
    public static void initResultsThread(){
        AppExecutors executorsInstance = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getAppExecutorsInstance();
        executorsInstance.diskIO().execute(new Runnable() {
            @Override
            public void run() {
                MathBrainerRepository repository = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getRepository();
                repository.initGameResults();
            }
        });
    }


    /**
     * ---------------------------------------------------------------------------------------------
     * Update game results in db
     * ---------------------------------------------------------------------------------------------
     */
    public static void incrementGameResultsThread(final String gameResultsName){
        AppExecutors executorsInstance = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getAppExecutorsInstance();
        executorsInstance.diskIO().execute(new Runnable() {
            @Override
            public void run() {
                MathBrainerRepository repository = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getRepository();
                repository.incrementGameResult(gameResultsName);
            }
        });
    }




    /**
     * ---------------------------------------------------------------------------------------------
     * Increment results for game which need to be incremented by more then 1 unit,
     * that is by a delta
     * ---------------------------------------------------------------------------------------------
     */
    public static void incrementGameResultByDeltaThread(final String gameResultsName, final int delta){
        AppExecutors executorsInstance = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getAppExecutorsInstance();
        executorsInstance.diskIO().execute(new Runnable() {
            @Override
            public void run() {
                MathBrainerRepository repository = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getRepository();
                repository.incrementGameResultByDelta(gameResultsName, delta);
            }
        });
    }



    /**
     * ---------------------------------------------------------------------------------------------
     * Update highscore for a game result if last score is greater
     * ---------------------------------------------------------------------------------------------
     */
    public static void updateGameResultHighscoreThread(final String gameResultsName, final int lastScore){
        AppExecutors executorsInstance = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getAppExecutorsInstance();
        executorsInstance.diskIO().execute(new Runnable() {
            @Override
            public void run() {
                MathBrainerRepository repository = ((SingletonProvider) SingletonProvider.Companion.getsContext()).getRepository();
                repository.updateGameResultHighscore(gameResultsName, lastScore);
            }
        });
    }










}

File: app/src/main/java/eu/indiewalkabout/mathbrainer/statistics/Results.kt
package eu.indiewalkabout.mathbrainer.statistics

import eu.indiewalkabout.mathbrainer.util.AppExecutors
import eu.indiewalkabout.mathbrainer.util.SingletonProvider

object Results {

    /**
     * ---------------------------------------------------------------------------------------------
     * Init game results in db
     * ---------------------------------------------------------------------------------------------
     */
    fun initResultsThread() {
        val executorsInstance = (SingletonProvider.getsContext() as SingletonProvider).appExecutorsInstance
        executorsInstance!!.diskIO().execute {
            val repository = (SingletonProvider.getsContext() as SingletonProvider).repository
            repository?.initGameResults()
        }
    }


    /**
     * ---------------------------------------------------------------------------------------------
     * Update game results in db
     * ---------------------------------------------------------------------------------------------
     */
    fun incrementGameResultsThread(gameResultsName: String) {
        val executorsInstance = (SingletonProvider.getsContext() as SingletonProvider).appExecutorsInstance
        executorsInstance!!.diskIO().execute {
            val repository = (SingletonProvider.getsContext() as SingletonProvider).repository
            repository?.incrementGameResult(gameResultsName)
        }
    }


    /**
     * ---------------------------------------------------------------------------------------------
     * Increment results for game which need to be incremented by more then 1 unit,
     * that is by a delta
     * ---------------------------------------------------------------------------------------------
     */
    fun incrementGameResultByDeltaThread(gameResultsName: String, delta: Int) {
        val executorsInstance = (SingletonProvider.getsContext() as SingletonProvider).appExecutorsInstance
        executorsInstance!!.diskIO().execute {
            val repository = (SingletonProvider.getsContext() as SingletonProvider).repository
            repository?.incrementGameResultByDelta(gameResultsName, delta)
        }
    }


    /**
     * ---------------------------------------------------------------------------------------------
     * Update highscore for a game result if last score is greater
     * ---------------------------------------------------------------------------------------------
     */
    fun updateGameResultHighscoreThread(gameResultsName: String, lastScore: Int) {
        val executorsInstance = (SingletonProvider.getsContext() as SingletonProvider).appExecutorsInstance
        executorsInstance!!.diskIO().execute {
            val repository = (SingletonProvider.getsContext() as SingletonProvider).repository
            repository?.updateGameResultHighscore(gameResultsName, lastScore)
        }
    }


}
